hedera-node/src/main/java/com/hedera/services/store/schedule/HederaScheduleStore.java

```diff
@@ -231,4 +232,21 @@ private void buildTxToEntityIdMap(Supplier<FCMap<MerkleEntityId, MerkleSchedule>
 
 		return Optional.empty();
 	}
+
+	@Override
+	public ResponseCodeEnum execute(ScheduleID id) {
+		var idRes = resolve(id);
+		if (idRes == MISSING_SCHEDULE) {
+			return INVALID_SCHEDULE_ID;
+		}
+
+		var schedule = get(id);
+		if (schedule.isDeleted()) {
+			return SCHEDULE_WAS_DELETED;
+		}
+
+		apply(id, DELETION);
+		txToEntityId.remove(new CompositeKey(Arrays.hashCode(schedule.transactionBody()), schedule.payer().toGrpcAccountId()));
```

Comment:
I think that we are constructing the `CompositeKey` in many places. Wouldn't it be better if we have `computeCompositeKey` method or smth like that and have only 1 place where `CompositeKey` instances are created.